corehq/apps/ota/views.py

```diff
@@ -139,22 +139,55 @@ def claim(request, domain):
     as_user_obj = CouchUser.get_by_username(as_user) if as_user else None
     restore_user = get_restore_user(domain, request.couch_user, as_user_obj)
 
-    case_id = unquote(request.POST.get('case_id', ''))
-    if not case_id:
+    # This will be expecting space seperated values of all the cases to be claimed
+    case_ids = unquote(request.POST.get('case_id', '')).split()
+
+    if not case_ids:
         return HttpResponse('A case_id is required', status=400)
 
-    try:
-        if get_first_claim(domain, restore_user.user_id, case_id):
-            return HttpResponse('You have already claimed that {}'.format(request.POST.get('case_type', 'case')),
-                                status=409)
-
-        claim_case(domain, restore_user, case_id,
-                   host_type=unquote(request.POST.get('case_type', '')),
-                   host_name=unquote(request.POST.get('case_name', '')),
-                   device_id=__name__ + ".claim")
-    except CaseNotFound:
-        return HttpResponse('The case "{}" you are trying to claim was not found'.format(case_id),
-                            status=410)
+    if len(case_ids) == 1:
+        try:
+            if get_first_claim(domain, restore_user.user_id, case_ids[0]):
+                return HttpResponse('You have already claimed that {}'.format(request.POST.get('case_type', 'case')),
+                                    status=409)
+
+            claim_case(domain, restore_user, case_ids[0],
+                    host_type=unquote(request.POST.get('case_type', '')),
+                    host_name=unquote(request.POST.get('case_name', '')),
+                    device_id=__name__ + ".claim")
+        except CaseNotFound:
+            return HttpResponse('The case "{}" you are trying to claim was not found'.format(case_ids[0]),
+                                status=410)
+    else:
+        case_ids_to_claim = list()
+        case_ids_already_claimed = list()
+        case_ids_not_found = list()
+
+        for case_id in case_ids:
```

Comment:
Given that this is the only place `get_first_claim` function is called, instead of looping and hitting the db for each case, I'd rework it to to exactly what you need here: accept a list of cases and return info about which ones already have claims. That could look like:
* Pass in a list of case ids instead of a single case id
* Update [this](https://github.com/dimagi/commcare-hq/blob/57de91beaefe553bfe0a006b9923cdc696d08fa4/corehq/ex-submodules/casexml/apps/case/cleanup.py#L143) to use `get_cases` instead of `get_case`
* Update the function to return the list of case ids that already have claims (or the case ids that don't have claims), since that's all you need to know.
* Rename the function to reflect its new behavior